src/components/app.jsx

```diff
@@ -49,6 +49,24 @@ import { useReducerWithThunk, reducer, initialState } from "../reducer.js";
 const _ = cockpit.gettext;
 const N_ = cockpit.noop;
 
+const MaybeBackdrop = ({ children }) => {
+    const [hasDialogOpen, setHasDialogOpen] = useState(false);
+
+    useEffect(() => {
+        window.addEventListener("message", event => {
```

Comment:
I think that you need to remove this listener when the component unmounts, i.e., return a function in this hook that cleans it.

Reply:
Good point, removing unused listeners prevents memory leaks. If this unmounts though it really means the the installer finished, so the user rebooted or so, so there is not big deal there.

I will add it still, does not hurt.